decentraland/screens/ListLandForSaleScreen.js

```diff
@@ -8,40 +8,72 @@ import {
 import PropTypes from "prop-types";
 import LandForSaleList from "@presentational/LandForSaleList";
 import LandForSaleListItem from "@presentational/LandForSaleListItem";
-import { addressesAreEqual, showError, showInfo, getContracts } from "@helpers";
+import { showError, showInfo, getContracts, addressesAreEqual } from "@helpers";
+import { generateAssetFromId } from "@helpers/decentraland";
 import { Root } from "native-base";
-
 import DecentralandUtils from "tasit-sdk/dist/helpers/DecentralandUtils";
 
-import AssetTypes from "@constants/AssetTypes";
-const { ESTATE, PARCEL } = AssetTypes;
-
 export class ListLandForSaleScreen extends React.Component {
   componentDidMount = async () => {
+    const {
+      appendLandForSaleToList,
+      setLoadingAssetsForSaleInProgress,
+    } = this.props;
     try {
       showInfo("Loading land for sale...");
-      await this._loadAssetsForSale();
+      const assetsForSale = await this._getAllAssetsForSale();
+
+      // Non-functional style
+      // const loadingAssetsOnScreen = [];
+      // for (let promise of assetsForSale) {
+      //   let loadAssetToScreen = async () => {
+      //     const assetForSale = await promise;
```

Comment:
Wouldn't awaiting it here not add it to the array until this line completes, which would mean losing the parallelism?

Reply:
I'm almost sure that won't. Following commented code how I understand it:

```javascript
// For each promise that will return an assetForSale...
const loadingAssetsOnScreen = assetsForSale.map(promise => {
  // Create a new function that returns a promise...
  let loadAssetOnScreen = async () => {
    const assetForSale = await promise;
    // to call the action function, the promise should be resolved
    appendLandForSaleToList(assetForSale);
  };
  // ... or in other words, warp a promise with another promise

   // returns a promise without awaiting it
  return loadAssetOnScreen();
});
// loadingAssetsOnScreen now has a list of promises (same as assetsForSale)
// and those promises will be waited by Promise.all()
```

Reply:
Wait, but there's no for loop in your commented code. I agree that the map ones should work as you describe.

The line I commented had a for loop.

Reply:
This discussion doesn't gate merging though, since we're going the map route

Reply:
> Wait, but there's no for loop in your commented code. I agree that the map ones should work as you describe.
> 
> The line I commented had a for loop.

Oh, you are right, my fault, I've thought that you are talking about the chosen style code. I think that the behavior will be the same on the `for` loop since the `await` is inside of the function and this function isn't being resolved inside of the loop. What do you think?

Reply:
Yeah, I suppose because it's wrapped in that `let loadAssetToScreen = async () => {` it will work fine. I was thinking of a for loop where the first thing that happened was awaiting with the `async` scope outside of the toop. Resolving this.